examples/ShowPlayer.cpp

```diff
@@ -66,90 +69,205 @@ int ShowPlayer::Init() {
+      entries[entry.universe] = entry;
+    }
+    if (!found && playhead_time == seek_time) {
+      // Gather frames from other universes before sending if landing on the
+      // trailing edge of a frame's timeout.
+      found = true;
+    } else if ((found && entry.next_wait > 0) || playhead_time > seek_time) {
+      break;
+    }
   }
+  m_playback_pos = playhead_time;
+
+  // Send data in the state it would be in at the given time
+  map<unsigned int, ShowEntry>::iterator entry_it;
+  for (entry_it = entries.begin(); entry_it != entries.end(); ++entry_it) {
+    SendFrame(entry_it->second);
+  }
+  // Adjust the timeout to handle landing in the middle of the entry's timeout
+  RegisterNextTimeout(playhead_time-seek_time);
+
+  return ShowLoader::OK;
 }
 
 
 /**
- * Get the next time offset
+ * Send the next frame in the show file
  */
-ShowLoader::State ShowPlayer::RegisterNextTimeout() {
-  unsigned int timeout;
-  ShowLoader::State state = m_loader.NextTimeout(&timeout);
-  if (state != ShowLoader::OK) {
-    return state;
+void ShowPlayer::SendNextFrame() {
+  ShowEntry entry;
+  ShowLoader::State state = m_loader.NextEntry(&entry);
+  const bool stop_point_hit = m_stop > 0 && m_playback_pos >= m_stop;
+  if (state == ShowLoader::END_OF_FILE || stop_point_hit) {
+    if (m_playback_pos == m_stop) {
+      // Send the last frame before looping/exiting
+      SendFrame(entry);
+    }
+    HandleEndOfFile();
+    return;
+  } else if (state == ShowLoader::INVALID_LINE) {
+    HandleInvalidLine();
+    return;
   }
+  SendEntry(entry);
+}
+
+
+/**
+ * Send @p entry, update playhead position, and wait for next
+ * @param entry the show file entry to send
+ */
+void ShowPlayer::SendEntry(const ShowEntry &entry) {
+  // Send DMX data
+  SendFrame(entry);
+
+  // Set when next to send data
+  m_playback_pos += entry.next_wait;
+  RegisterNextTimeout(entry.next_wait);
+}
 
+
+/**
+ * Send the next frame in @p timeout milliseconds
+ */
+void ShowPlayer::RegisterNextTimeout(const unsigned int timeout) {
   OLA_INFO << "Registering timeout for " << timeout << "ms";
   m_client.GetSelectServer()->RegisterSingleTimeout(
-      timeout,
-      ola::NewSingleCallback(this, &ShowPlayer::SendNextFrame));
-  return state;
+     timeout,
+     ola::NewSingleCallback(this, &ShowPlayer::SendNextFrame));
+}
+
+
+/**
+ * Send data contained in @p entry
+ */
+void ShowPlayer::SendFrame(const ShowEntry &entry) const {
+  if (entry.buffer.Size() == 0) {
+    return;
+  }
+  OLA_INFO << "Universe: " << entry.universe << ": " << entry.buffer.ToString();
+  ola::client::SendDMXArgs args;
+  m_client.GetClient()->SendDMX(entry.universe, entry.buffer, args);
 }
 
 
 /**
  * Handle the case where we reach the end of file
  */
 void ShowPlayer::HandleEndOfFile() {
-  m_iteration_remaining--;
-  if (m_infinite_loop || m_iteration_remaining > 0) {
-    m_loader.Reset();
+  if (m_iteration_remaining > 0) {
+    m_iteration_remaining--;
```

Comment:
Do we print the remaining iteration count elsewhere? It would be nice to see somewhere.